scripts/vaccination_alerts/send-emails.ts

```diff
@@ -0,0 +1,206 @@
+import _ from 'lodash';
+import admin from 'firebase-admin';
+import { getFirestore } from '../common/firebase';
+import EmailService, {
+  EmailSendData,
+  isInvalidEmailError,
+} from '../alert_emails/email-service';
+import * as yargs from 'yargs';
+import { DEFAULT_ALERTS_FILE_PATH, readVaccinationAlerts } from './utils';
+import { RegionVaccinePhaseInfo } from '../../src/cms-content/vaccines/phases';
+
+const BATCH_SIZE = 20;
+
+class SendVaccinationAlertsService {
+  db: FirebaseFirestore.Firestore;
+  emailService: EmailService;
+
+  invalidEmailCount: number = 0;
+  errorCount: number = 0;
+  emailSent: number = 0;
+
+  uniqueEmailAddress: { [email: string]: number } = {};
+  locationsWithEmails: { [fips: string]: number } = {};
+
+  constructor(db: FirebaseFirestore.Firestore, emailService: EmailService) {
+    this.db = db;
+    this.emailService = emailService;
+  }
+
+  private async onEmailSent(email: string, alert: RegionVaccinePhaseInfo) {
+    this.updateSentEmailCounters(email, alert);
+    const docPath = `${alert.fips}/emailVersions/${alert.emailAlertVersion}/emails/${email}`;
+    return this.db
+      .collection('vaccination-alerts')
+      .doc(docPath)
+      .set({ sentAt: admin.firestore.FieldValue.serverTimestamp() });
+  }
+
+  private async onInvalidEmail(email: string) {
+    this.invalidEmailCount += 1;
+    const querySnapshot = await this.db
+      .collection('alerts-subscriptions')
+      .doc(email)
+      .get();
+    const currentData = querySnapshot.data();
+    if (currentData) {
+      await this.db
+        .collection('invalid-alert-subscriptions')
+        .doc(email)
+        .set(currentData);
+      await this.db.collection('alerts-subscriptions').doc(email).delete();
+    }
+  }
+
+  private updateSentEmailCounters(
+    email: string,
+    alert: RegionVaccinePhaseInfo,
+  ) {
+    this.emailSent += 1;
+    this.uniqueEmailAddress[email] = (this.uniqueEmailAddress[email] || 0) + 1;
+    // Todo don't make just fips
+    const key = (alert.fips, alert.emailAlertVersion);
```

Comment:
Uhh...  This looks like it's trying to make a tuple, but that's not a JavaScript feature, so I think it's equivalent to `const key = alert.emailAlertVersion` which is almost certainly not what we want? 😁 

easiest bet is probably `const key = `${alert.fips}_${alert.emailAlertVersion}` ?